test/integration/image-optimizer/test/index.test.js

```diff
@@ -0,0 +1,282 @@
+  })
+
+  it('should resize relative url and wildcard accept header as webp', async () => {
+    const query = { url: '/test.png', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/*' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/webp')
+  })
+
+  it('should resize absolute url from localhost', async () => {
+    const url = `http://localhost:${appPort}/test.png`
+    const query = { url, w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/webp' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/webp')
+  })
+
+  it('should use cached image file when parameters are the same', async () => {
+    const query = { url: '/test.png', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/webp' } }
+
+    const res1 = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res1.status).toBe(200)
+    expect(res1.headers.get('Content-Type')).toBe('image/webp')
+    const json1 = await fsToJson(imagesDir)
+    expect(json1).toBeTruthy()
+
+    const res2 = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res2.status).toBe(200)
+    expect(res2.headers.get('Content-Type')).toBe('image/webp')
+    const json2 = await fsToJson(imagesDir)
+    expect(json2).toStrictEqual(json1)
+  })
+
+  it('should proxy-pass unsupported image types and should not cache file', async () => {
+    const json1 = await fsToJson(imagesDir)
+    expect(json1).toBeTruthy()
+
+    const query = { url: '/test.bmp', w: 64, q: 80 }
+    const opts = { headers: { accept: 'image/invalid' } }
+    const res = await fetchViaHTTP(appPort, '/_next/image', query, opts)
+    expect(res.status).toBe(200)
+    expect(res.headers.get('Content-Type')).toBe('image/bmp')
+
+    const json2 = await fsToJson(imagesDir)
+    expect(json2).toStrictEqual(json1)
+  })
+}
+
+describe('Image Optimizer', () => {
+  describe('dev support', () => {
+    beforeAll(async () => {
+      appPort = await findPort()
+      app = await launchApp(appDir, appPort)
+    })
+    afterAll(async () => {
+      await killApp(app)
+      await fs.remove(imagesDir)
+    })
+
+    runTests()
+  })
+
+  describe('Server support', () => {
+    beforeAll(async () => {
+      await nextBuild(appDir)
+      appPort = await findPort()
+      app = await nextStart(appDir, appPort)
+    })
+    afterAll(async () => {
+      await killApp(app)
+      await fs.remove(imagesDir)
+    })
+
+    runTests()
+  })
+
+  describe('Serverless support', () => {
+    beforeAll(async () => {
+      const target = 'target: "serverless",'
+      const [first, ...rest] = nextConfigContent.split('\n')
+      const content = [first, target, ...rest].join('\n')
+      await fs.writeFile(nextConfig, content)
```

Comment:
This can be simplified using the `File` class mentioned above example here https://github.com/vercel/next.js/blob/5b1be2bb980c4cad7e543a9f150c1a1c50a6c0f3/test/integration/i18n-support/test/index.test.js#L790
```suggestion
      nextConfig.replace('// target', 'target')
```